### PR TITLE
Adds initial env vars for commonly used services

### LINKED ISSUE #49: Add env variables for common plugins and services
We tend to use a lot of the same plugins and services — S3, Imgix, CloudFront, etc. — on a lot of our projects. I think expanding `.env.example` to have vars for these things will:

- serve as sort of a checklist when getting started
- help our naming be a bit more consistent between projects

```env
# AWS S3 settings
S3_ACCESS_KEY_ID=
S3_SECRET_ACCESS_KEY=
S3_BUCKET_NAME=
S3_REGION=
S3_SUBFOLDER=

# CloudFront settings
CLOUDFRONT_DOMAIN=
CLOUDFRONT_DISTRIBUTION_ID=
CLOUDFRONT_PATH_PREFIX=

# IMGIX settings
IMGIX_API_KEY=
IMGIX_SECURITY_TOKEN=
IMGIX_SOURCE_DOMAIN=
```

### PR DESCRIPTION
**Pull request by chrismlusk**
- Organizes into settings groups
- Adds variables for S3, CloudFront, Imgix
- Closes #49

### CODE REVIEW

File: .env.example [hunk 1]
```diff
@@ -1,42 +1,39 @@
...
-
-# (Optional) Instead of setting DB_DRIVER, DB_SERVER, DB_USER, DB_PASSWORD, DB_DATABASE, and DB_PORT
-# you can put all of these into one URL.
-# DATABASE_URL="mysql://user@pass:my.example.com:3306/database_name"
+DB_PORT="3306"
+# (Optional) Instead of separate vars, you can put them into one URL
+# DB_DSN="mysql://user@pass:my.example.com:3306/database_name"
+
+# URL and path settings
+PUBLIC_URL="https://example.test"
```
[the remark this answers is not in the document]
Comment (chrismlusk) [reply .env.example:20]: Agreed! My secret plan was to get this approved, merge it into the Gulp feature branch, and suggest we change it there. So... now it's no longer a secret.

**Review by brianjhanson — COMMENTED** [review]
While we're in here, we should remove quotes from all the values

File: .env.example [hunk 2]
```diff
@@ -1,42 +1,38 @@
...
+DB_SERVER=127.0.0.1
+DB_USER=mysqluser
+DB_PASSWORD=mysqlpass
+DB_DATABASE=
+DB_SCHEMA=public
+DB_TABLE_PREFIX=
+DB_PORT=3306
+# (Optional) Instead of separate vars, you can put them into one URL
+# DB_DSN=mysql://user@pass:my.example.com:3306/database_name
+
```
Comment (flynnt) [inline .env.example:18]: I kinda feel like we should have an opinion about using the `DB_DSN` or not. I actually like to use it, but I'm fine with whatever as long as we pick one.

File: .env.example [hunk 3]
```diff
@@ -1,42 +1,38 @@
...
+# URL and path settings
+PRIMARY_SITE_URL=https://example.test
+
+# S3 settings
+S3_ACCESS_KEY_ID=
+S3_SECRET_ACCESS_KEY=
+S3_BUCKET_NAME=
+S3_REGION=
```
Comment (brianjhanson) [inline .env.example:26]: Setting up Senda today and realized that I rarely use these too and instead use the Craft UI for it

File: .env.example (hunk 2 above)
Comment (chrismlusk) [reply .env.example:18]: Are there any benefits to `dsn`? I think a list of values is easier to read than a single line, but that's the extent of my preference here.

Comment (brianjhanson) [reply .env.example:18]: I would vote for our usual. [There was a long thread on the Craft repo](https://github.com/craftcms/cms/issues/6159) about this, and they eventually ended up on the old config, but DSN takes precedence

**Comment by brianjhanson** [comment]
I think this is common knowledge, but in case it's not. The `./craft setup` command will step you through creating the `.env` file. Generally that's what I use to start a new project rather than copying this file.

File: .env.example (hunk 2 above)
Comment (flynnt) [reply .env.example:18]: So maybe let's keep the usual and maybe remove the `DB_DSN` comment? Or may that's Craft's job? I dunno, it always seems a little silly to have like a "hey by the way" for a value you already set.

File: .env.example (hunk 3 above)
Comment (chrismlusk) [reply .env.example:26]: Yeah, that's probably normal. Our current recommendation is to use the same bucket for staging and production, but I do like having them all _available_ as env vars if or when you need them.

**Comment by chrismlusk** [comment]
True! This example file is more like documentation or a resource to help you down the path ODC typically takes on projects.

File: .env.example (hunk 3 above)
Comment (flynnt) [reply .env.example:26]: I tend to use these values. While not S3, this came up on Poetry where the buckets change per environment. I think if you use the UI, it saves the bucket name as a string to the config, which would mess with any setup that had a per-environment bucket situation.

File: .env.example (hunk 2 above)
Comment (brianjhanson) [reply .env.example:18]: I would take 100% of Craft's [`.env.example`](https://github.com/craftcms/craft/blob/master/.env.example) and then just add our special stuff, so no DSN in that case

Comment (chrismlusk) [reply .env.example:16]: Done.

File: .env.example (hunk 3 above)
Comment (chrismlusk) [reply .env.example:24]: Let's keep them. This is just an example file and nobody is forced to use them, but IMO it's good idea to use them so when a per-environment scenario does arise, you're ready to go.

**Pull request merged**